internal/service/runners/consumers/mint.go

```diff
@@ -0,0 +1,244 @@
+	}
+
+	openseaData := opensea.Metadata{
+		Name:        fmt.Sprintf("%s #%d", task.Attributes.TokenName, task.Attributes.TokenId),
+		Description: book.Data.Attributes.Description,
+		Image:       c.ipfsLoader.BaseUri + task.Attributes.BannerIpfsHash,
+		FileURL:     fileLink.Data.Attributes.Url,
+	}
+	// Uploading metadata
+	if err = c.ipfsLoader.UploadMetadata(openseaData); err != nil {
+		return errors.Wrap(err, "failed to load metadata to the ipfs")
+	}
+
+	// Uploading banner
+	if err = c.ipfsLoader.UploadBanner(task.Attributes.BannerIpfsHash); err != nil {
+		return errors.Wrap(err, "failed to load banner to the ipfs")
+	}
+
+	taskStatus := coreResources.TaskFinishedUploading
+	if err := c.core.UpdateTask(coreModels.UpdateTaskParams{
+		Id:     cast.ToInt64(task.ID),
+		Status: &taskStatus,
+	}); err != nil {
+		return errors.Wrap(err, "failed to update task`s marketplace id and status")
+	}
+	return nil
+}
+
+func (c *MarketPlaceConsumer) MintUpdating(task coreResources.Task, event etherdata.TokenSuccessfullyPurchasedEvent) error {
+	if _, err := c.database.Payments().New().Insert(data.Payment{
+		ContractAddress:   event.ContractAddress.String(),
+		TokenId:           task.Attributes.TokenId,
+		BookId:            task.Attributes.BookId,
+		PayerAddress:      event.Recipient.String(),
+		NftId:             event.NftId,
+		TokenAddress:      event.Erc20Info.TokenAddress.String(),
+		TokenSymbol:       event.Erc20Info.Symbol,
+		TokenName:         event.Erc20Info.Name,
+		TokenDecimals:     event.Erc20Info.Decimals,
+		Amount:            event.Amount.String(),
+		PriceMinted:       event.MintedTokenPrice.String(),
+		PriceToken:        event.PaymentTokenPrice.String(),
+		BannerLink:        c.ipfsLoader.BaseUri + task.Attributes.BannerIpfsHash,
+		PurchaseTimestamp: event.Timestamp,
+		ChainId:           c.network.ChainId,
+		Type:              int8(event.Type),
+	}); err != nil {
+		return errors.Wrap(err, "failed to add payment to the table")
+	}
+
+	return nil
+}
+
+func (c *MarketPlaceConsumer) UpdateStatistics(book bookerModels.GetBookResponse, event etherdata.TokenSuccessfullyPurchasedEvent) error {
+	pricePerOneToken := new(big.Float).Quo(new(big.Float).SetInt(event.PaymentTokenPrice), big.NewFloat(math.Pow10(18)))
+	paymentTokenAmount := new(big.Float).Quo(new(big.Float).SetInt(event.Amount), big.NewFloat(math.Pow10(18)))
+	usdPrice := big.NewFloat(0).Mul(paymentTokenAmount, pricePerOneToken)
+	return c.database.KeyValue().UpdateStatistics(
+		// Amount
+		data.KeyValue{ // amount of each book
+			Key:   "stats-book-" + book.Data.ID + "-amount",
+			Value: "1",
+		}, data.KeyValue{ // amount of all books
+			Key:   "stats-book-amount-total",
+			Value: "1",
+		}, data.KeyValue{ // amount of all books by day
+			Key:   "stats-book-" + book.Data.ID + "-amount-date-" + time.Now().Format("2006-Jan-02"),
+			Value: "1",
+		},
+
+		// Price
+		// Native Currency
+		data.KeyValue{ // price (native currency) by each book by each token
```

Comment:
Is it really a good idea to store everything on the KV table? What if I want to analyze statistics and filter by, say, date? Filtering by "like" is a very bad idea. You need a separate table for that, and think through the structure of tables.

Reply:
Will be fix on feature/statistic branch